datalad/distributed/ora_remote.py

```diff
@@ -498,18 +560,21 @@ def get(self, src, dst, progress_cb):
                     progress_cb(bytes_received)
 
     def rename(self, src, dst):
-        self._run('mv {} {}'.format(sh_quote(str(src)), sh_quote(str(dst))))
+        with self.ensure_writeable(dst.parent):
+            self._run('mv {} {}'.format(sh_quote(str(src)), sh_quote(str(dst))))
 
     def remove(self, path):
         try:
-            self._run('rm {}'.format(sh_quote(str(path))))
+            with self.ensure_writeable(path.parent):
+                self._run('rm {}'.format(sh_quote(str(path))), check=True)
         except RemoteCommandFailedError as e:
-            raise RIARemoteError(
-                "Write permissions for a key's parent"
-                "directory are also required to drop content.") from e
+            raise RIARemoteError("Write permissions for a key's parent "
+                                 "directory are also required to drop "
+                                 "content.") from e
```

Comment:
Keeping this message unchanged is now confusing. The problem is no longer that there are no write permissions. The problem is that write permissions could not be obtained. This can happen when the path is owned by another user, for example. This should be corrected. Right now it would error like so:

```
Write permissions for a key's parent directory are also required to drop content. caused by [Errno 1] Operation not permitted: '/data/project/ukb_vbm/outputstore/893/8de76-0302-45b5-9825-3c6ce3f3fffe/annex/objects/j6/Pq/MD5E-s45608--c22377262d535937cc36659c70bb36a3.tar.gz'
```

the shown path is the parent directory.

Reply:
Right, thx. Will fix.

Reply:
Pushed a changed message.